fairscale/nn/misc/checkpoint_activations.py

```diff
@@ -63,25 +64,55 @@ def checkpoint_wrapper(module: nn.Module, offload_to_cpu: bool = False) -> nn.Mo
         (nn.Module):
             wrapped module
     """
+    # Build a list of BN classes and use two functions only make them track
+    # running stats in one of the forward function.
+    bn_list = []
+    for name, child in module.named_modules():
+        # _BatchNorm is base for bn1d, bn2d, bn3d and sync_bn, apex_sync_bn
+        if isinstance(child, _BatchNorm):
+            bn_list.append([child, None])
+
+    def pre_forward() -> None:
+        for bn_and_bool in bn_list:
+            bn_and_bool[1] = bn_and_bool[0].track_running_stats  # type: ignore
+            bn_and_bool[0].track_running_stats = False  # type: ignore
+
+    def post_forward() -> None:
+        for bn_and_bool in bn_list:
+            assert bn_and_bool[1] is not None
+            bn_and_bool[0].track_running_stats = bn_and_bool[1]  # type: ignore
+
     # The use of weakref here is to prevent creating a ref cycle: m -> m.forward -> m.
     # When such cycle exists, gc won't collect the module when the module is freed.
     # That causes GPU memory to be leaked. See the unit test for how we catch that.
     #
     # We prefer this over a class wrapper since the class wrapper would have to
     # proxy a lot of fields and methods.
-    module.forward = functools.partial(_checkpointed_forward, type(module).forward, weakref.ref(module), offload_to_cpu)  # type: ignore
+    module.forward = functools.partial(  # type: ignore
+        _checkpointed_forward, type(module).forward, weakref.ref(module), offload_to_cpu, pre_forward, post_forward
+    )
```

Comment:
can we directly `register_forward_pre_hook` and `register_forward_hook` on the module instead of piping these through the checkpointing method?